src/components/giving-tuesday/Countdown.js

```diff
@@ -1,19 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
 
 import CountdownPie from './CountdownPie';
 import UseCountdown from '../../hooks/useCountdown';
 
+const getDescriptorString = (descriptor, time) => {
+  return time !== 1 ? `${descriptor}s` : descriptor;
+};
+
 const Countdown = () => {
-  /**
-   * manually set date to 11/28/2023 w/ a UTC offset (no daylight savings)
-   */
-  const targetDate = new Date(Date.UTC(2023, 10, 29, 7, 59, 59));
-  const [days, hours, minutes, seconds] = UseCountdown(targetDate);
+  const [countdownDate, setCountdownDate] = useState(null);
+  const [days, hours, minutes, seconds] = UseCountdown(countdownDate) || [];
+
+  useEffect(() => {
+    if (!countdownDate) {
+      /**
+       * manually set date to 04/04/2024 7AM w/ a UTC offset (no daylight savings)
+       */
+      setCountdownDate(new Date(Date.UTC(2024, 3, 4, 14, 0, 0)));
+    }
+  }, [countdownDate, setCountdownDate]);
```

Comment:
@wmui51,

I'm not sure I understand why the date needs to be set to a state on the component mount. Is it going to change while the user is interacting with the page? 

If you're planning on making this configurable in Storyblok, I still think you can statically set the date in the component after passing in the date through a prop.

Reply:
Ah right, we accounted for the offset by setting the date to UTC values but we then ran into the issue of the first render showing the countdown time from when the website was rendered. That first render caused a brief flash with an incorrect amount of time remaining before showing the correct time.

Ok, I think I've caught up again. Thanks.

Reply:
Yah, it will still blink empty initially which I'm trying to work on fixing, but the main fix is if the countdown rings change like last time and there is a weird re-render mismatch.

Thanks for approving.

Reply:
@sherakama has this been deployed to dev as well?

Reply:
This code is available for you on dev now.